acceptance/tests/sameness/sameness_test.go

```diff
@@ -261,145 +270,354 @@ func TestFailover_Connect(t *testing.T) {
+							// 2. A workaround to perform the DNS and PQ queries on the primary datacenter cluster by specifying the admin partition
+							// e.g kubectl --context kind-dc1 --namespace ns1 exec -i deploy/static-client -c static-client \
+							//	-- dig @test-3lmypr-consul-dns.default static-server.service.ns2.ns.mine.sg.ap1.ap.consul
+							// is not possible at the moment due to a bug. The workaround will be used once this bug is fixed.
+							logger.Logf(t, "skipping DNS and PQ checks for %s", tt.name)
+						}
+
+						// Scale down static-server on the current failover, will fail over to the next.
+						logger.Logf(t, "scaling server down on %s", v.failoverServer.name)
+						k8s.KubectlScale(t, v.failoverServer.serverOpts, staticServerDeployment, 0)
+					}
+				})
+			}
 		})
 	}
 }
 
-type member struct {
+type expectedPQ struct {
+	partition string
+	peerName  string
+	namespace string
+}
+
+type cluster struct {
+	name           string
+	partition      string
 	context        environment.TestContext
 	helmCluster    *consul.HelmCluster
 	client         *api.Client
 	hasServer      bool
 	serverOpts     *terratestk8s.KubectlOptions
 	clientOpts     *terratestk8s.KubectlOptions
 	staticServerIP *string
+	pqID           *string
+	pqName         *string
+	dnsIP          *string
+	acceptors      []string
+}
+
+type clusters map[string]*cluster
+
+func (c clusters) resetScale(t *testing.T) {
+	for _, v := range c {
+		k8s.KubectlScale(t, v.serverOpts, staticServerDeployment, 1)
+	}
+}
+
+// setServerIP makes sure everything is up and running and then saves the
+// static-server IP to the appropriate cluster. IP addresses can change when
+// services are scaled up and down.
+func (c clusters) setServerIP(t *testing.T) {
+	for _, labelSelector := range []string{"app=static-server", "app=static-client"} {
+		for k, v := range c {
+			podList, err := v.context.KubernetesClient(t).CoreV1().Pods(metav1.NamespaceAll).List(context.Background(),
+				metav1.ListOptions{LabelSelector: labelSelector})
+			require.NoError(t, err)
+			require.Len(t, podList.Items, 1)
+			require.Len(t, podList.Items[0].Spec.Containers, 2)
+			if labelSelector == "app=static-server" {
+				ip := &podList.Items[0].Status.PodIP
+				require.NotNil(t, ip)
+				logger.Logf(t, "%s-static-server-ip: %s", v.name, *ip)
+				c[k].staticServerIP = ip
+			}
+		}
+	}
+}
+
+// verifyServerUpState will verify that the static-servers are all up and running as
+// expected by curling them from their local datacenters.
+func (c clusters) verifyServerUpState(t *testing.T) {
+	logger.Logf(t, "verifying that static-servers are up")
+	for _, v := range c {
+		// Query using a client and expect its own name, no failover should occur
+		serviceFailoverCheck(t, v, v.name)
+	}
+}
+
+func copySecret(t *testing.T, cfg *config.TestConfig, sourceContext, destContext environment.TestContext, secretName string) {
```

Comment:
Would any of these helpers be useful in the framework?

Reply:
hesitant for copySecret because you don't always want to delete the secret like this

Reply:
yeah, I think putting these in the framework would confuse things with the cleanups, you have to think about what you need the behaviour to be